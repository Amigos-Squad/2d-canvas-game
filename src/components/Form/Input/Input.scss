@import 'scss/var';

.input {
  display: flex;
  flex-direction: column;
  gap: 2px 0;

  &__label {
    display: flex;
    font-size: 1rem;
    font-weight: bold;
    color: $font-color;
    margin-left: 1px;
  }

  &__field {
    border: none;
    outline: none;
    border-radius: 4px;
    color: $font-color;
    height: 40px;
    font-size: 20px;
    padding-left: 8px;
    background-color: $bg-light-color;
    width: 100%;
  }

  &__error {
    color: red;
  }

  &_horizontal {
    flex-direction: row;
  }

  &_horizontal &__label {
    display: flex;
    min-width: 8vw;
    align-items: flex-end;
    text-transform: uppercase;
    font-size: 1.1vw;
    line-height: 40px;
  }

  &_horizontal &__field {
    align-items: flex-end;
  }
}
