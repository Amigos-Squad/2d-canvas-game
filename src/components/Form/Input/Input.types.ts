export interface InputRule {
  message: string,
  rule: RegExp
}

export type Props = {
  onChange: (e: React.FormEvent<HTMLInputElement>) => void;
  value: string;
  horizontal?: boolean;
  rules?: InputRule[]
} & React.HTMLProps<HTMLInputElement>;
