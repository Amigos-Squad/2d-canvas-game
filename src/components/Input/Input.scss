@import 'scss/var';

.input {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;

  &__label {
    font-size: 0.8rem;
    font-weight: bold;
    color: $font-color;
  }

  &__field {
    border: none;
    outline: none;
    color: $font-color;
    height: 40px;
    background-color: $bg-light-color;
  }
}
