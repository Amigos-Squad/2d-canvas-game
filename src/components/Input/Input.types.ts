export type Props = {
    onChange: (e: React.FormEvent<HTMLInputElement>) => void;
    value: string;
} & React.HTMLProps<HTMLInputElement>;
