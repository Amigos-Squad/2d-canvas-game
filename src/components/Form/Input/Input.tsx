import React, { ReactElement, memo } from 'react';
import './Input.scss';
import { Props } from './Input.types';

export const Input = memo(
  ({ onChange, value, label, horizontal, ...rest }: Props): ReactElement => (
    <label className={`input ${horizontal ? 'input_horizontal' : ''}`}>
      <div className="input__label">{label}</div>
      <input
        className="input__field"
        onChange={onChange}
        value={value}
        {...rest}
      />
    </label>
  )
);
