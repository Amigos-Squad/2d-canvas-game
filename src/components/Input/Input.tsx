import React, { ReactElement, memo } from 'react';
import './Input.scss';
import { Props } from './Input.types';

export const Input = memo(
  ({ onChange, value, label, ...rest }: Props): ReactElement => (
    <div className="input">
      <label className="input__label">{label}</label>
      <input
        className="input__field"
        onChange={onChange}
        value={value}
        {...rest}
      />
    </div>
  )
);
