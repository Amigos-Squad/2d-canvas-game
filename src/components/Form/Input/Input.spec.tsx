import TestRenderer from 'react-test-renderer';
import { Input } from './Input';
import React from 'react';

describe('Компонент ввода', () => {
    it('снапшот поля ввода', () => {
        const tree = TestRenderer.create(<Input value="" onChange={() => undefined} label="Тест"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
