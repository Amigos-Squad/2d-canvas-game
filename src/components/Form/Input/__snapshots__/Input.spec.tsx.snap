// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент ввода снапшот поля ввода 1`] = `
<label
  className="input"
>
  <div
    className="input__label"
  >
    Тест
  </div>
  <div
    className="input__container"
  >
    <input
      className="input__field"
      onChange={[Function]}
      value=""
    />
    <div
      className="input__error"
    >
      
    </div>
  </div>
</label>
`;
